pkg/codeconfig/server.go

```diff
@@ -88,6 +89,24 @@ func (s *Server) Declare(ctx context.Context, req *pb.ResourceDeclareRequest) (*
 	return &pb.ResourceDeclareResponse{}, nil
 }
 
+func (s *Server) Details(ctx context.Context, req *pb.ResourceDetailsRequest) (*pb.ResourceDetailsResponse, error) {
+	switch req.Resource.Type {
+	case pb.ResourceType_Api:
+		return &pb.ResourceDetailsResponse{
+			Provider: "dev",
+			Service:  "Api",
+			Id:       req.Resource.Name,
+			Details: &pb.ResourceDetailsResponse_Api{
+				Api: &pb.ApiResourceDetails{
+					Url: "http://localhost:50051/apis/" + req.Resource.Name,
+				},
+			},
+		}, nil
+	default:
+		return nil, fmt.Errorf("unsupported resource type %s", req.Resource.Type)
+	}
+}
```

Comment:
As mentioned by @asalkeld this will prevent config collection from failing if this method is included in inline code (not in a runtime context).

example:
```typescript
const publicApi = api('public');
// Will normally fail at config collection
const deets = publicApi.details();
```
> Note in runtimes like cloud run this may also fail to deploy at runtime as the function will exist before the API gateway is deployed.

Providing mock details to avoid this error might also cause issues though and doesn't really help the user with what the issue actually is:

```typescript
const publicApi = api('public');
// Will normally fail at config collection
const deets = publicApi.details();

// user then does something with details outside of runtime
// this will still fail at config collection time
const result = fetch(deets.url);
```

Ignoring the deployment race condition for Cloud Run, we should educate and provide methods that make it clear how to exclude runtime code from being executing at deploy time. This can currently be done with env variable checks, but we should provide more convenience than that.

```
const dbClient =  atRuntime(() => {
  const client = new DbClient(connectionDetails);
  runMigrations(client);
  return client;
});
```
> This is just an example, any other suggestions/ideas that are better would be great!